src/main/java/examples/Examples.java

```diff
@@ -410,4 +410,24 @@ public void example12(MongoService mongoService) {
 
   }
 
+  public void example13_0(MongoService mongoService) {
+
+    JsonObject document = new JsonObject().put("title", "The Hobbit")
+        .put("publicationDate", new JsonObject().put("$date", -1018659600000L /* 21/9/1937 */));
```

Comment:
* If I remember well, I tried to use `java.util.Calendar` or `java.util.Date` to get the long timestamp in the example : both failed at build time because of code generation (I think). Any idea about this? I agree that `-1018659600000L` is ugly :(

* The spec (http://docs.mongodb.org/manual/reference/mongodb-extended-json/) is the mongo REST API & shell spec. Vert.x uses the java mongo driver. The method [org.bson.BsonWriter#writeDateTime](https://github.com/mongodb/mongo-java-driver/blob/master/bson/src/main/org/bson/BsonWriter.java#L67) takes a long. And `org.bson.BsonReader#readDateTime` returns a long also. I chose to stay as close as possible to the driver behaviour. So I think I could say I used the *shell mode* of the spec :). 

If we follow strictly the REST API spec, then we need to use ISO-8601 date format instead of long. It means parsing the input and format the output in `io.vertx.ext.mongo.impl.codec.json.JsonObjectCodec` (and also on client side, when dealing with `JsonObject`). In any case, I think it's not a good idea to use the `"$numberLong"` workaround. 

What do you think?